src/nft/components/NFTMint.tsx

```diff
@@ -0,0 +1,64 @@
+import { background, border, cn, color } from '../../styles/theme';
+import { useIsMounted } from '../../useIsMounted';
+import { useTheme } from '../../useTheme';
+import { LifecycleType, type NFTMintReact } from '../types';
+import { NFTLifecycleProvider } from './NFTLifecycleProvider';
+import { NFTMintProvider } from './NFTMintProvider';
+import { NFTProvider } from './NFTProvider';
+
+export function NFTMint({
+  children,
+  className,
+  contractAddress,
+  tokenId,
+  useNFTData,
+  useNFTMintData,
+  buildMintTransaction,
+  onStatus,
+  onError,
+  onSuccess,
+}: NFTMintReact) {
+  const componentTheme = useTheme();
+
+  const isMounted = useIsMounted();
```

Comment:
do we need this in both NFTMint and NFTView component ? or could we move to wrapper ?

Reply:
there's not really any wrapper for these, they are just separate top level components.

Reply:
yes, i think we can, let me work something up

Reply:
I consolidated the styling and these two checks to a single NFT component used in both cards.  I also removed the NFTMintProvider allowing one data hook to be passed and defined which data applies to each component in types.  The NFTMint card will be required when using the MintButton as it sets lifecycle status separately and allows a user to pass in the buildTransaction function.